lib/inv_ca_cg.cpp

```diff
@@ -53,6 +51,153 @@ namespace quda {
     if (!param.is_preconditioner) profile.TPSTOP(QUDA_PROFILE_FREE);
   }
 
+  CACGNE::CACGNE(DiracMatrix &mat, DiracMatrix &matSloppy, SolverParam &param, TimeProfile &profile) :
+    CACG(mmdag, mmdagSloppy, param, profile), mmdag(mat.Expose()), mmdagSloppy(matSloppy.Expose()),
+    xp(nullptr), yp(nullptr), init(false) {
+  }
+
+  CACGNE::~CACGNE() {
+    if ( init ) {
+      if (xp) delete xp;
+      if (yp) delete yp;
+      init = false;
+    }
+  }
+
+  // CACGNE: M Mdag y = b is solved; x = Mdag y is returned as solution.
+  void CACGNE::operator()(ColorSpinorField &x, ColorSpinorField &b) {
+    if (param.maxiter == 0 || param.Nsteps == 0) {
+      if (param.use_init_guess == QUDA_USE_INIT_GUESS_NO) blas::zero(x);
+      return;
+    }
+
+    const int iter0 = param.iter;
+
+    if (!init) {
+      ColorSpinorParam csParam(x);
+      csParam.create = QUDA_NULL_FIELD_CREATE;
+      xp = ColorSpinorField::Create(x, csParam);
+      csParam.create = QUDA_ZERO_FIELD_CREATE;
+      yp = ColorSpinorField::Create(x, csParam);
+      init = true;
+    }
+
+    double b2 = blas::norm2(b);
+
+    if (param.use_init_guess == QUDA_USE_INIT_GUESS_YES) {
+
+      // compute initial residual
+      mmdag.Expose()->M(*xp,x);
+      double r2 = blas::xmyNorm(b,*xp);
```

Comment:
clang-format suggested the following change:


```
-      mmdag.Expose()->M(*xp,x);
-      double r2 = blas::xmyNorm(b,*xp);
+      mmdag.Expose()->M(*xp, x);
+      double r2 = blas::xmyNorm(b, *xp);
```

<!-- identifier: clang-format -->
<!-- File: lib/inv_ca_cg.cpp -->